src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -346,17 +346,9 @@ private unsafe T[] Populate(string? name, MemberListType listType, CacheType cac
                             // allocating on the stack is faster than allocating on the GC heap
                             // but we surely don't want to cause a stack overflow
                             // no one should be looking for a member whose name is longer than 1024
-                            if (cUtf8Name > MAXNAMELEN)
+                            Span<byte> utf8Name = (uint)cUtf8Name > MAXNAMELEN ? new byte[cUtf8Name] : stackalloc byte[cUtf8Name];
+                            fixed (byte* pUtf8Name = utf8Name)
                             {
-                                byte[] utf8Name = new byte[cUtf8Name];
-                                fixed (byte* pUtf8Name = &utf8Name[0])
-                                {
-                                    list = GetListByName(pName, cNameLen, pUtf8Name, cUtf8Name, listType, cacheType);
-                                }
-                            }
-                            else
-                            {
-                                byte* pUtf8Name = stackalloc byte[cUtf8Name];
                                 list = GetListByName(pName, cNameLen, pUtf8Name, cUtf8Name, listType, cacheType);
```

Comment:
Does GetListByName still need to accept pointers?

Reply:
Addressed! Although, Filter and its MdUtf8String still expect pointers, I'll look into cleaning it up separately